src/components/ClaimAdministratorFormPanel.js

```diff
@@ -14,35 +14,55 @@ const styles = (theme) => ({
 const ClaimAdministratorFormPanel = (props) => {
   const { edited, classes, modulesManager, onEditedChanged, readOnly } = props;
   const { formatMessage } = useTranslations("admin.ClaimAdministratorFormPanel", modulesManager);
-  const isEnabled = edited.userTypes?.includes(CLAIM_ADMIN_USER_TYPE);
-
-  const has_role = !!edited.roles ? (edited.roles.filter((x) => x.isSystem == 256).length != 0) : false
-  if (isEnabled) {
-    const {isLoading, data, error: graphqlError} = useGraphqlQuery(
-      `
-      query UserRolesPicker ($system_id: Int) {
-        role(systemRoleId: $system_id) {
-          edges {
-            node {
-              id name isSystem
-            }
+  const [isEnabled, setIsEnabled] = useState(false);
+  const hasClaimUserType = edited.userTypes?.includes(CLAIM_ADMIN_USER_TYPE);
+  const hasClaimRole = edited.roles ? edited.roles.filter((x) => x.isSystem === 256).length !== 0 : false;
+  const {
+    isLoading,
+    data,
+    error: graphqlError,
+  } = useGraphqlQuery(
+    `
+    query UserRolesPicker ($system_id: Int) {
+      role(systemRoleId: $system_id) {
+        edges {
+          node {
+            id name isSystem
           }
         }
       }
-    `,
-      { system_id: 256 }, // Claim Admin System Role is 256
-    );
-    const isValid = !isLoading
-    useEffect(() => {
-      if (isValid & isEnabled & !has_role) {
-        const role = data.role.edges[0].node
-        const roles  = !!edited.roles ? edited.roles : [];
-        roles.push(role)
-        edited.roles = roles
-        onEditedChanged(edited)
+    }
+  `,
+    { system_id: 256 }, // Claim Admin System Role is 256
+  );
+  const isValid = !isLoading;
+
+  useEffect(() => {
+    const roles = edited?.roles ?? [];
+    const claimRole = data?.role.edges[0].node;
+
+    if (isValid && isEnabled && !hasClaimRole) {
+      roles.push(claimRole);
+      edited.roles = roles;
+      onEditedChanged(edited);
+    } else if (isValid && !isEnabled) {
+      const filteredRoles = roles.filter((role) => role.isSystem !== 256);
+      edited.roles = filteredRoles;
+      onEditedChanged(edited);
+    }
+  }, [isEnabled]);
+
+  useEffect(() => {
```

Comment:
@olewandowski1 Same as in previous comment - looks like this useEffect is quite similar  as in '[EnrolmentOfficerFormPanel.js](https://github.com/openimis/openimis-fe-admin_js/pull/35/files#diff-dc8704c04aa3f4a8c4f2f8a9fe8d9cc46076651bf0b6e67d48f726447f89e9c3)'.
Can we create separate function to use it within those useEffects? I think this could be easy when it comes to the parametrize.

Reply:
Done.